### ISSUE TITLE
Rules ignored when asserting nested arrays with keyNested

### ISSUE THREAD

**Issue by mscreenie** [Respect/Validation#689]
Tested with post data:
```
Array
(
    [organisation] => Array
        (
            [name] => f
            [address] => 3r3wrr3r
        )

    [contact] => Array
        (
            [email] => ffesfewf2232313123212
            [password] => 
            [name] => wdwfwef
            [position] => wdwdwf
            [number] => £"!@£;21#£:"!:£~!":£'21;
        )
```

Rules
```
        v::keyNested('organisation.name', v::length(2, 50)->notEmpty())
            ->keyNested('organisation.address', v::length(2, 50)->notEmpty())
            ->keyNested('contact.name', v::length(1, 50)->notEmpty())
            ->keyNested('contact.email', v::email()->notEmpty())
            ->keyNested('contact.password', v::length(3, 100)->notEmpty())
            ->keyNested('contact.position', v::length(1, 100)->notEmpty())
            ->keyNested('contact.number', v::phone()->notEmpty())
            ->validate($_POST["config"]["init"]["signup"]);
```

Being completely ignored and not throwing any exceptions, any ideas? I've tried removing notEmpty when using length.

**Comment by batusa** [comment]
to get an exception in case of failure you should use `check` or `assert` instead of `validate`.

more info here: http://respect.github.io/Validation/docs/ (_Exception types_ section).

**Comment by mscreenie** [comment]
Sorry, asset was my original use, I switched to validate before posting here.

I get a different problem when using assert. A expception is thrown by I cannot collect any information about the errors from findMessages

    } catch (NestedValidationException $exception) {

        $errors = $exception->findMessages([
            'organisation' => [
                'name' => 'Center name',
                'address' => 'Center address'
            ],
            'contact' => [
                'name' => 'Contact name',
                'password' => 'Contact name',
                'position' => 'Contact name',
                'number' => 'Contact name',
                'email' => 'Contact name'
            ]]);

print_r($errors);

    }

**Comment by batusa** [comment]
we use  `findMessages` when we want to select a specific *set/type* of rules or when we want to overwrite the default error messages.

please, look for  _Getting messages as an array by name_ and _Custom messages_ section on the docs for detailed examples.

there are other methods to get the error messages, for example: `getFullMessage` and `getMessages`.

**Comment by mscreenie** [comment]
Thanks, I'll use the other methods for now but if I may be frank, findMessages should work for nested arrays / keys just as much as anything else? I may be missing something here.

It makes it difficult passing back to client views in this case JS to match up against an element. I'm going to hope I can ->setName() against keyNested.

**Comment by mscreenie** [comment]
Is there any interest in changing findMessages to include nested keys at all? I would like to think I am not alone another issue is open about this https://github.com/Respect/Validation/issues/635

**Comment by henriquemoody** [comment]
This bug was finally fixed on versions [1.0.11](https://packagist.org/packages/respect/validation#1.0.11) and [1.1.6](https://packagist.org/packages/respect/validation#1.1.6) by #675.

I'm very sorry about my delay (I talked about that already https://github.com/Respect/Validation/issues/620#issuecomment-237474435).

Thanks for reporting! :panda_face:
